src/robot/ConcreteRobot.cpp

```diff
@@ -235,7 +235,8 @@ Eigen::VectorXd ConcreteRobot::getAccelerationLimits(
 
 // ==============================================================================
 CollisionFreePtr ConcreteRobot::getSelfCollisionConstraint(
-    const MetaSkeletonStateSpacePtr& space, const MetaSkeletonPtr& metaSkeleton)
+    const MetaSkeletonStateSpacePtr& space,
+    const MetaSkeletonPtr& metaSkeleton) const
```

Comment:
So as discussed with @gilwoolee, it is best if we have the function just return `const` [example, `ConstTestablePtr` but the constraint is being passed into the planning methods and constructors which take `TestablePtr`. Since we are changing the planning API anyway, I think it is best if we wait for the corrections to happen there and make the changes here post that. I'll make an issue if that's agreeable [ :+1: ].

Reply:
Sounds good to me! :+1: